tests/unit/test_diffsync.py

```diff
@@ -898,3 +884,147 @@ class NoDeleteInterfaceDiffSync(BackendA):
+                "name": "eth1",
+            },
+            "nyc-spine2__eth0": {
+                "description": "Interface 0",
+                "device_name": "nyc-spine2",
+                "name": "eth0",
+            },
+            "nyc-spine2__eth1": {
+                "description": "Interface 1",
+                "device_name": "nyc-spine2",
+                "name": "eth1",
+            },
+            "rdu-spine1__eth0": {
+                "description": "Interface 0",
+                "device_name": "rdu-spine1",
+                "name": "eth0",
+            },
+            "rdu-spine1__eth1": {
+                "description": "Interface 1",
+                "device_name": "rdu-spine1",
+                "name": "eth1",
+            },
+            "rdu-spine2__eth0": {
+                "description": "Interface 0",
+                "device_name": "rdu-spine2",
+                "name": "eth0",
+            },
+            "rdu-spine2__eth1": {
+                "description": "Interface 1",
+                "device_name": "rdu-spine2",
+                "name": "eth1",
+            },
+            "sfo-spine1__eth0": {
+                "description": "Interface 0",
+                "device_name": "sfo-spine1",
+                "name": "eth0",
+            },
+            "sfo-spine1__eth1": {
+                "description": "Interface 1",
+                "device_name": "sfo-spine1",
+                "name": "eth1",
+            },
+            "sfo-spine2__eth0": {
+                "description": "TBD",
+                "device_name": "sfo-spine2",
+                "name": "eth0",
+            },
+            "sfo-spine2__eth1": {
+                "description": "ddd",
+                "device_name": "sfo-spine2",
+                "name": "eth1",
+            },
+            "sfo-spine2__eth2": {
+                "description": "Interface 2",
+                "device_name": "sfo-spine2",
+                "name": "eth2",
+            },
+        },
+        "person": {"Glenn Matthews": {"name": "Glenn Matthews"}},
+        "site": {
+            "nyc": {"devices": ["nyc-spine1", "nyc-spine2"], "name": "nyc"},
+            "rdu": {
+                "devices": ["rdu-spine1", "rdu-spine2"],
+                "name": "rdu",
+                "people": ["Glenn Matthews"],
+            },
+            "sfo": {"devices": ["sfo-spine1", "sfo-spine2"], "name": "sfo"},
+        },
+    }
+    backend_a_by_dict = BackendA()
+    backend_a_by_dict.load_from_dict(data)
+    assert backend_a.dict() == backend_a_by_dict.dict()
+
+
+def test_diffsync_get_or_none(backend_a):
+    assert backend_a.get_or_none(backend_a.site, "rdu") is not None
+    assert backend_a.get_or_none(backend_a.site, "does-not-exist") is None
+
+
+def test_diffsync_get_value_order(backend_a):
+    assert backend_a._get_value_order() == [  # pylint: disable=protected-access
+        "site",
+        "unused",
+        "device",
+        "interface",
+        "person",
```

Comment:
This seems like a weird order to me, especially given that it doesn't appear to match the order that `get_tree_traversal()` prints?

Reply:
Perhaps more descriptive as `_get_initial_value_order`?

Reply:
updated.